Userland/Utilities/pathchk.cpp

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2021, the SerenityOS developers.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <AK/String.h>
+#include <LibCore/ArgsParser.h>
+#include <LibCore/File.h>
+#include <bits/posix1_lim.h>
+#include <unistd.h>
+
+int main(int argc, char** argv)
+{
+    bool fail = false;
+    static bool flag_most_posix = false;
+    static bool flag_portability = false;
+    static bool flag_empty_name_and_leading_dash = false;
+
+    if (pledge("stdio rpath", nullptr) < 0) {
+        perror("pledge");
+        return 1;
+    }
+
+    Vector<const char*> paths;
+
+    Core::ArgsParser args_parser;
+    args_parser.add_option(flag_most_posix, "Check for most POSIX systems", nullptr, 'p');
+    args_parser.add_option(flag_empty_name_and_leading_dash, "Check for empty names and leading dash", nullptr, 'P');
+    args_parser.add_option(flag_portability, "Check portability (equivalent to -p and -P)", "portability", '\0');
+    args_parser.add_positional_argument(paths, "Path to check", "path", Core::ArgsParser::Required::Yes);
+    args_parser.parse(argc, argv);
+
+    if (flag_portability) {
+        flag_most_posix = true;
+        flag_empty_name_and_leading_dash = true;
+    }
+
+    for (auto& path : paths) {
+        unsigned long path_max = flag_most_posix ? _POSIX_PATH_MAX : pathconf(path, _PC_PATH_MAX);
+        unsigned long name_max = flag_most_posix ? _POSIX_NAME_MAX : pathconf(path, _PC_NAME_MAX);
+
+        if (String(path).length() > path_max) {
```

Comment:
You are converting path to String in multiple places, do it once at the beginning of the loop?
Also does it really need to be a string, could we get away with StringView?

Reply:
> You are converting path to `String` in multiple places, do it once at the beginning of the loop?

Done.

> Also does it really need to be a `String`, could we get away with `StringView`?

Sometimes it needs to be a String.